### PR TITLE
linux: config: Add Intel E1000e

### PR DESCRIPTION
**Pull request by legoater**
The E1000 driver has reset issues on the QEMU PowerNV machine with POWER9 :

[   16.140708] Unable to handle kernel paging request for data at address 0xc00c000000000000
[   16.142203] Faulting instruction address: 0xc008000000aa9b0c
cpu 0x0: Vector: 300 (Data Access) at [c0000000f5833840]
    pc: c008000000aa9b0c: e1000_io_write+0x64/0x7c [e1000]
    lr: c008000000aac0f0: e1000_reset_hw+0x1bc/0x48c [e1000]
    sp: c0000000f5833ac0
   msr: 900000000280b033
   dar: c00c000000000000
 dsisr: 42000000
  current = 0xc0000000f579a200
  paca    = 0xc000000001ec0000	 irqmask: 0x03	 irq_happened: 0x01
    pid   = 13, comm = kworker/0:1

Include the E1000e as a replacement for such QEMU machines.

Signed-off-by: Cédric Le Goater <clg@kaod.org>

### CODE REVIEW

**Comment by shenki** [comment]
Can we remove `CONFIG_E1000`?

**Comment by shenki** [comment]
retest this please

**Comment by stewart-ibm** [comment]
I'm okay with merging when doc/boot-devices.rst is updated.

**Comment by legoater** [comment]
should be ok now.

File: doc/boot-devices.rst [hunk 1]
```diff
@@ -35,6 +35,7 @@ it from this table and notify the person mentioned in the owner field.
 +-------------------------------+-----------------------+--------+----------------------------+
 | Exar Xframe 10Gb              | S2IO                  |        |                            |
 +-------------------------------+-----------------------+--------+----------------------------+
+| Intel PRO/1000 PCIe           | E1000E                |        |                            |
 | Intel PRO/1000                | E1000                 | Qemu   | stewart@linux.vnet.ibm.com |
```
Comment (shenki) [inline doc/boot-devices.rst]: Is there a reason to keep the e1000 around?

**Comment by stewart-ibm** [comment]
retest this please

**Pull request merged**